sdk/lib/rtl/heapuser.c

```diff
@@ -127,3 +127,29 @@ RtlInitializeHeapManager(VOID)
     RtlInitializeCriticalSection(&RtlpProcessHeapsListLock);
 }
 
+ULONG
+NTAPI
+RtlpGetProcessHeaps(ULONG count,
+    HANDLE* heaps)
+{
+    PPEB Peb = RtlGetCurrentPeb();
+
+    RtlEnterCriticalSection(&RtlpProcessHeapsListLock);
+
+    ULONG nHeaps = Peb->NumberOfHeaps;
+    if (heaps)
+    {
+        for (ULONG n = 0; n < min(nHeaps, count); ++n)
+        {
+            heaps[n] = Peb->ProcessHeaps[n];
+        }
+    }
```

Comment:
We should probably guard this with SEH, and release the `RtlpProcessHeapsListLock` in a finally handler?

Reply:
Done since I had other changes to push as well.